Doc/library/typing.rst

```diff
@@ -36,6 +36,24 @@ New features are frequently added to the ``typing`` module.
 The `typing_extensions <https://pypi.org/project/typing-extensions/>`_ package
 provides backports of these new features to older versions of Python.
 
+Deprecation Timeline of Major Features
+======================================
+
+Certain features in ``typing`` are deprecated and may be removed in a future
+version of Python. The following table summarizes major deprecations for your
+convenience. This is subject to change, and not all deprecations are listed.
+
++----------------------------------+---------------+-------------------+----------------+
+|  Feature                         | Deprecated in | Projected removal | PEP/issue      |
++==================================+===============+===================+================+
+|  ``typing.io`` and ``typing.re`` | 3.8           | 3.12              | :issue:`38291` |
+|  submodules                      |               |                   |                |
++----------------------------------+---------------+-------------------+----------------+
+|  ``typing`` versions of standard | 3.9           | 3.14              | :pep:`585`     |
```

Comment:
This will need discussion in an issue, but I would prefer that we do not carry out this removal. It will cause a lot of churn for users and I see little benefit in removing these names. We can minimize their presence (e.g., make the documentation mention them only minimally; change the runtime code to just `List = list`), but I don't want to remove them.

Reply:
Agreed. I'll remove them for now.

Reply:
Hmm, I've instead decided to leave it there, but remove the version to represent that we haven't decided/will not actually remove it.

Reply:
There is another Note: box mentioning the deprecation of the Genetic builtin aliases here https://docs.python.org/3/library/typing.html#module-contents
and that one actually says they will be removed in the first python 5 years after the release of 3.9.0
Maybe that should be removed, otherwise the "not planned" and that line will be contradictory and confusing.

Reply:
I'd like for us to get rid of these objects from `typing` eventually. But I agree with Jelle that we should go for a very long deprecation period,  much like we did for the aliases `collections` had to objects in `collections.abc`.

Anyway, Jelle's correct that we should probably be discussing that in an issue rather than here 😀

Reply:
A more complete review to come when I arrive in SLC; this remark comes to you from the Minneapolis airport.